modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/impl/CassandraExperimentRepository.java

```diff
@@ -1194,4 +1234,35 @@ public void createApplication(Application.Name applicationName) {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Map<Application.Name, Set<String>> getTagListForApplications(Collection<Application.Name> applicationNames) {
+
+        LOGGER.debug("Retrieving Experiment Tags for applications {}", applicationNames);
+
+        try {
+            List<ListenableFuture<Result<ExperimentTagsByApplication>>> futures = new ArrayList<>();
+
+            for (Application.Name appName : applicationNames) {
```

Comment:
@AndreaSuckro I understand @bargenilesh suggested that the IN query in Cassandra is not efficient but probably @bargenilesh can confirm whether it is true for small number of rows as well (as you had rightly asked earlier). If it is inefficient for such small rows as well, then probably instead of hitting cassandra with a query for each application, it would be better to just fetch all the rows (which is expected to be in the order of 100s or worst case 1000s which should still be ok) and do an in-memory filtering (within our code) for applications we need. That would be much faster and cleaner.